### ISSUE TITLE
zod() Incompatible with z.ZodType

### ISSUE THREAD

**Issue by rogerfar** [ciscoheat/sveltekit-superforms#369]
I have this schema:

```
export type User = {
  userId: string;
  name: string;
  manager?: User | null;
}

export const UserSchema: z.ZodType<User> = z.object({
  userId: z.string().uuid(),
  name: z.string(),
  manager: z.union([z.null(), z.lazy(() => UserSchema)]),
});

```

I have to use `z.ZodType<User>` otherwise TS static type checking fails.

But when I do:
```
await superValidate(user, zod(UserSchema));
```

TypeScript throws an error:
```
Argument of type 'ZodType<User, ZodTypeDef, User>' is not assignable to parameter of type 'ZodValidation<AnyZodObject | ZodObjectUnion<AnyZodObject>>'.
  Property 'removeDefault' is missing in type 'ZodType<User, ZodTypeDef, User>' but required in type 'ZodDefault<ZodEffects<ZodEffects<ZodEffects<ZodEffects<ZodEffects<ZodEffects<ZodEffects<AnyZodObject | ZodObjectUnion<AnyZodObject>, { [x: string]: any; }, { [x: string]: any; } | undefined>, { ...; }, { ...; } | undefined>, { ...; }, { ...; } | undefined>, { ...; }, { ...; } | undefined>, { ...; }, { ...; } | undef...'.
```

It runs fine though, it looks just to be a typing issue.

**Comment by ciscoheat** [comment]
Should be easy to add to the accepted schema types in the next release.

**Comment by ciscoheat** [comment]
Should be fixed now with 2.7.0, can you confirm that it is?

**Comment by rogerfar** [comment]
Looks good here!
